addons/sale/models/account_move.py

```diff
@@ -28,6 +28,21 @@ def action_post(self):
                 pass
         return res
 
+    def button_draft(self):
+        #inherit of the function from account.move to validate a new tax and the priceunit of a downpayment
+        res = super(AccountMove, self).button_draft()
+        line_ids = self.mapped('line_ids').filtered(lambda line: line.sale_line_ids.is_downpayment)
+        for line in line_ids:
+            try:
+                line.sale_line_ids.tax_id = line.tax_ids
```

Comment:
this was already set on the `action_post`, it seems to be untouched in the mentioned flow so it's not needed the reassignment